### PR TITLE
net-dns/knot: version bump

### PR DESCRIPTION
**Pull request by nemunaire**
Nothing special in this release.

```diff
--- knot-2.7.4.ebuild	2018-12-10 12:22:05.148469892 +0100
+++ knot-2.7.6.ebuild	2019-01-30 20:17:05.581000105 +0100
@@ -1,4 +1,4 @@
-# Copyright 1999-2018 Gentoo Authors
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
```

### CODE REVIEW

File: net-dns/knot/knot-2.7.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		dev-libs/protobuf-c
+	)
+	geoip? ( dev-libs/libmaxminddb )
+	idn? (
+		!libidn2? ( net-dns/libidn:* )
+		libidn2? ( net-dns/libidn2 )
+	)
+	systemd? ( sys-apps/systemd )
+"
+BDEPEND="${RDEPEND}
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: I'm quite certain those `RDEPEND` should not be in `BDEPEND`. Instead, it seems your ebuild is missing `DEPEND`.

Comment (nemunaire) [reply net-dns/knot/knot-2.7.6.ebuild]: Thanks for the review @a17r, I think I misunderstood the changes in EAPI 7. `RDEPEND`s are needed for the `configure` step.

Comment (a17r) [reply net-dns/knot/knot-2.7.6.ebuild]: Yes, so they should end up in `DEPEND` as well.

Comment (nemunaire) [reply net-dns/knot/knot-2.7.6.ebuild]: Right, thanks for the pointer. I just update the commit. I left only `virtual/pkgconfig` and `dev-python/sphinx` which are used on the build-host side.

File: net-dns/knot/knot-2.7.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	dev-python/lmdb
+	net-libs/gnutls
+	caps? ( sys-libs/libcap-ng )
+	dnstap? (
+		dev-libs/fstrm
+		dev-libs/protobuf-c
+	)
+	geoip? ( dev-libs/libmaxminddb )
+	idn? (
+		!libidn2? ( net-dns/libidn:* )
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: are you sure this is correct? `net-dns/libidn` has subslots and is apparently source/binary slotted (`0` vs. `1.33`), so if it links to libidn you will want to trigger rebuild on subslot bump.

File: net-dns/knot/knot-2.7.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		libidn2? ( net-dns/libidn2 )
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: are you sure this is correct? `net-dns/libidn2` has subslots defined, so if it links to libidn2 you will want to trigger rebuild on subslot bump.

File: net-dns/knot/knot-2.7.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	)
+	systemd? ( sys-apps/systemd )
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: If it links to `libudev` or `libsystemd` you will likely want to trigger rebuild here as well on subslot (=ABI) break.

File: net-dns/knot/knot-2.7.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	dev-libs/libedit
+	dev-libs/userspace-rcu:=
+	dev-python/lmdb
+	net-libs/gnutls
+	caps? ( sys-libs/libcap-ng )
+	dnstap? (
+		dev-libs/fstrm
+		dev-libs/protobuf-c
+	)
+	geoip? ( dev-libs/libmaxminddb )
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: `dev-libs/libmaxminddb` has a subslot defined, please check if your package requires rebuild on subslot bump of this dep.

File: net-dns/knot/knot-2.7.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+RDEPEND="
+	dev-db/lmdb
+	dev-libs/libedit
+	dev-libs/userspace-rcu:=
+	dev-python/lmdb
+	net-libs/gnutls
+	caps? ( sys-libs/libcap-ng )
+	dnstap? (
+		dev-libs/fstrm
+		dev-libs/protobuf-c
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: `dev-libs/protobuf-c` has a subslot defined, please check if your package requires rebuild on subslot bump of this dep.

File: net-dns/knot/knot-2.7.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	"+whoami"
+)
+IUSE="doc caps +fastparser idn libidn2 systemd +utils ${KNOT_MODULES[@]}"
+
+RDEPEND="
+	dev-db/lmdb
+	dev-libs/libedit
+	dev-libs/userspace-rcu:=
+	dev-python/lmdb
+	net-libs/gnutls
```
Comment (a17r) [inline net-dns/knot/knot-2.7.6.ebuild]: `net-libs/gnutls` has a subslot defined, please check if your package requires rebuild on subslot bump of this dep.

**Comment by a17r** [comment]
Maybe an opportunity to also drop one or two old versions?

**Comment by a17r** [comment]
ping

**Comment by nemunaire** [comment]
Here is the diff since the previous push, including changes you requested:

```diff
--- a/net-dns/knot/knot-9999.ebuild
+++ b/net-dns/knot/knot-9999.ebuild
@@ -26,25 +26,25 @@ KNOT_MODULES=(
        "+synthrecord"
        "+whoami"
 )
-IUSE="doc caps +fastparser idn libidn2 systemd +utils ${KNOT_MODULES[@]}"
+IUSE="doc caps +fastparser idn +libidn2 systemd +utils ${KNOT_MODULES[@]}"
 
 RDEPEND="
        dev-db/lmdb
        dev-libs/libedit
        dev-libs/userspace-rcu:=
        dev-python/lmdb
-       net-libs/gnutls
+       net-libs/gnutls:=
        caps? ( sys-libs/libcap-ng )
        dnstap? (
                dev-libs/fstrm
-               dev-libs/protobuf-c
+               dev-libs/protobuf-c:=
        )
-       geoip? ( dev-libs/libmaxminddb )
+       geoip? ( dev-libs/libmaxminddb:= )
        idn? (
-               !libidn2? ( net-dns/libidn:* )
-               libidn2? ( net-dns/libidn2 )
+               !libidn2? ( net-dns/libidn:0= !net-dns/libidn2 )
+               libidn2? ( net-dns/libidn2:= )
        )
-       systemd? ( sys-apps/systemd )
+       systemd? ( sys-apps/systemd:= )
 "
 DEPEND="${RDEPEND}"
 BDEPEND="
```

**Pull request closed without merging**